idioms/mem-replace.md

```diff
@@ -36,6 +36,28 @@ fn a_to_b(e: &mut MyEnum) {
 }
 ```
 
+This also works with more variants:
+
+```Rust
+use std::mem;
+
+enum MultiVariateEnum {
+    A { name: String },
+    B { name: String },
+    C,
+    D
+}
+
+fn swizzle(e: &mut MultiVariateEnum) {
```

Comment:
Could you add a comment explaining what this function does and what mem::replace is being used for? This seems somewhat confusing since the code would have the same observable behaviour without the mem::replace.